dvc/command/init.py

```diff
@@ -56,4 +59,13 @@ def add_parser(subparsers, parent_parser):
             "This operation removes local cache."
         ),
     )
+    init_parser.add_argument(
+        "--sub",
```

Comment:
How about
```suggestion
        "--subrepo",
```
then to match config?

Reply:
I agree, though I would stay with `subrepo` for user convenience

Reply:
@jorgeorpinel Would really appreciate your thoughts :slightly_smiling_face:

Reply:
Thanks for asking! I would avoid "repo" if it doesn't imply a Git repo. We use the term "DVC project" more generally. `--subproject`? Or copy Git (`submodule`/`subtree`)

Why do you need to use an option though? Can't DVC recognize it's already inside a DVC project? 🙂

Reply:
worried about `submodule` being confused with git submodules :slightly_frowning_face: `--subproject` seems alright, though not ideal too... Maybe there is a way to emphasize that this is option is to just tell dvc to use parent dir git...

> Why do you need to use an option though? Can't DVC recognize it's already inside a DVC project? slightly_smiling_face

Sure it can, but it can't understand that it needs to grab a git from an outside directory, this needs to be explicit.

This might be a point where `--no-scm` comes to bite us in the bum. If we would've made it explcit like `dvc config core.scm false`, then we could totally automatically use parent git here. Though that might also be error-prone, as some unsuspecting user might run it in one of his subdirs accidentally...

Reply:
OK. Not sure I understood completely, but anyway: if this option only applies to Git enabled DVC projects then `--subrepo` is a good name.

Reply:
Continued in https://github.com/iterative/dvc/pull/3257. Resolving here.